### PR TITLE
crimson: tmap support, list_snaps support, fixes for zero and cmpext

### PR DESCRIPTION
**Pull request by athanatos**
<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by athanatos** [comment]
This fixes several test_librbd tests.  Next up are various snapshot related hangs.

**Comment by athanatos** [comment]
jenkins test make check

**Comment by tchaikov** [comment]
jenkins test make check

**Comment by athanatos** [comment]
jenkins test make check

**Comment by athanatos** [comment]
jenkins test make check

**Comment by athanatos** [comment]
jenkins test make check

File: src/crimson/osd/ops_executer.cc [hunk 1]
```diff
@@ -484,6 +484,75 @@ OpsExecuter::call_errorator::future<> OpsExecuter::do_assert_ver(
...
+      auto p = ss.clone_overlap.find(clone);
+      if (p == ss.clone_overlap.end()) {
+	logger().error(
+	  "OpsExecutor::do_list_snaps: {} has inconsistent "
+	  "clone_overlap, missing clone {}",
+	  os.oi.soid,
+	  clone);
+	return crimson::ct_error::invarg::make();
+      }
+      ci.snaps.reserve(p->second.num_intervals());
```
Comment (liu-chunmei) [inline src/crimson/osd/ops_executer.cc:522]: should be ci.overlap.reserve?

Comment (athanatos) [reply src/crimson/osd/ops_executer.cc]: Yep, good catch!

**Comment by liu-chunmei** [comment]
jenkins test make check

**Review by liu-chunmei — APPROVED** [review]

**Pull request merged**